core/src/main/java/org/apache/carbondata/core/util/CarbonUtil.java

```diff
@@ -3231,4 +3231,42 @@ private static int unsetLocalDictForComplexColumns(List<ColumnSchema> allColumns
     }
     return columnLocalDictGenMap;
   }
+
+  /**
+   * This method get the carbon file format version
+   *
+   * @param carbonTable
+   * carbon Table
+   */
+  public static ColumnarFormatVersion getFormatVersion(CarbonTable carbonTable) throws IOException
+  {
+    String tablePath = carbonTable.getTablePath();
+    CarbonFile[] carbonFiles = FileFactory
+        .getCarbonFile(tablePath)
+        .listFiles(new CarbonFileFilter() {
+          @Override
+          public boolean accept(CarbonFile file) {
+            if (file == null) {
+              return false;
+            }
+            return file.getName().endsWith("carbonindex");
```

Comment:
I think it is better to get the version from data file instead of index file which is an optional file

Reply:
@ravipesala Is there any utility func to get the version from file footer already?

Reply:
It should not be under carbontable object. It is file version and it is related to each load, not table.  @manishgupta88 I think your PR already handles the validation of datamap loading on old formats right?

Reply:
@ravipesala ...If dataMap is created on legacy store then in my PR I am throwing exception at the time block/blocklet pruning...but the dataMap creation blocking is not taken care in my PR...

Reply:
so how can we get the version info for a table?